setup/src/Magento/Setup/Mvc/Bootstrap/InitParamListener.php

```diff
@@ -97,14 +94,25 @@ public function onBootstrap(MvcEvent $e)
     }
 
     /**
-     * @inheritdoc
+     * Create service. Proxy to the __invoke method
+     *
+     * @deprecared use the __invoke method instead
      *
      * @param ServiceLocatorInterface $serviceLocator
-     * @return mixed
+     * @return array
+     * @throws \Interop\Container\Exception\ContainerException
      */
     public function createService(ServiceLocatorInterface $serviceLocator)
     {
-        return $this->extractInitParameters($serviceLocator->get('Application'));
+        return $this($serviceLocator, 'Application');
+    }
```

Comment:
I believe that we may just replace this method with `__invoke` instead of extend the class by the new method.
@sivaschenko please, correct me if I'm not right

Reply:
I'd prefer to keep the existing method for backward compatibility purposes

Reply:
I agree, but it's setup sources. I'm not sure that someone extends or customizes the setup process.

Reply:
Yes, you are probably right